index.js

```diff
@@ -1,52 +1,77 @@
-import Listr from "listr";
+import Listr from 'listr';
 
-import commandLineArgs from "command-line-args";
-import { migrateSchema } from "./tasks/schema.js";
-import { migrateFiles } from "./tasks/files.js";
-import { migrateUsers } from "./tasks/users.js";
-import { migrateData } from "./tasks/data.js";
+import commandLineArgs from 'command-line-args';
+import { migrateSchema } from './tasks/schema.js';
+import { migrateFiles } from './tasks/files.js';
+import { migrateUsers } from './tasks/users.js';
+import { migrateData } from './tasks/data.js';
+import * as fs from 'fs';
 
 const commandLineOptions = commandLineArgs([
-	{
-		name: "skipCollections",
-		alias: "s",
-		type: String,
-		multiple: true,
-		defaultValue: [],
-	},
+  {
+    name: 'skipCollections',
+    alias: 's',
+    type: String,
+    multiple: true,
+    defaultValue: [],
+  },
 ]);
 
 const tasks = new Listr([
-	{
-		title: "Migrating Schema",
-		task: (context) => {
-			context.skipCollections = commandLineOptions.skipCollections;
-			return migrateSchema(context);
-		},
-	},
-	{
-		title: "Migration Files",
-		task: migrateFiles,
-	},
-	{
-		title: "Migrating Users",
-		task: migrateUsers,
-	},
-	{
-		title: "Migrating Data",
-		task: migrateData,
-	},
+  //   {
+  //     title: 'Migrating Schema',
+  //     task: context => {
+  //       context.skipCollections = commandLineOptions.skipCollections;
+  //       return migrateSchema(context);
+  //     },
+  //   },
+  //   {
+  //     title: 'Migration Files',
+  //     task: migrateFiles,
+  //   },
+  //   {
+  //     title: 'Migrating Users',
+  //     task: migrateUsers,
+  //   },
+  {
+    title: 'Setup Context',
+    task: setupContext,
+  },
+  {
+    title: 'Migrating Data',
+    task: migrateData,
+  },
+  //   {
+  //     title: 'Write Context',
+  //     task: writeContext,
+  //   },
 ]);
 
+async function writeContext(context) {
```

Comment:
it was helpful writing the context to a file after every step. Then I was able to see what it was trying to do, and maintain state between runs. Then I could repopulate the context and pick up where I left off.